src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VCFComparator.java

```diff
@@ -0,0 +1,887 @@
+            }
+        }
+        final VariantContextBuilder vcBuilder = new VariantContextBuilder(variant);
+        final List<Allele> orderedRelevantAlleles = new ArrayList<>(relevantAlleles);
+        if (vcBuilder.getGenotypes().size() == 1 && !ignoreNonRefData) {
+            orderedRelevantAlleles.add(Allele.NON_REF_ALLELE);
+        }
+        vcBuilder.alleles(orderedRelevantAlleles);
+        //subset allele-specific annotations
+        //final List<Map<String,Object>> annotations = GATKVariantContextUtils.splitAttributesIntoPerAlleleLists(variant, Collections.emptyList(), getHeaderForVariants());
+        //final int[] keepAlleleIndices = isSingleSample ? AlleleSubsettingUtils.getIndexesOfRelevantAllelesForGVCF(variant.getAlleles(), orderedRelevantAlleles, variant.getStart(), variant.getGenotype(0), false) : AlleleSubsettingUtils.getIndexesOfRelevantAlleles()
+
+
+        //NOTE that we use BEST_MATCH_TO_ORIGINAL for post-reblocked VCFs with no hom ref PLs
+        final GenotypesContext gc = AlleleSubsettingUtils.subsetAlleles(variant.getGenotypes(), variant.getGenotype(0).getPloidy(),  //mixed ploidy will be a problem, but we don't do that in practice
+                variant.getAlleles(), orderedRelevantAlleles, null, GenotypeAssignmentMethod.BEST_MATCH_TO_ORIGINAL);
+        vcBuilder.genotypes(gc);
+
+        final Map<String,Object> subsetAnnotations = ReblockGVCF.subsetAnnotationsIfNecessary(annotatorEngine, false, VCFConstants.GENOTYPE_POSTERIORS_KEY, variant, vcBuilder.make(), Collections.emptyList());
+
+        return variant.getGenotype(0).isHomRef() ? vcBuilder.make() : GATKVariantContextUtils.reverseTrimAlleles(vcBuilder.attributes(subsetAnnotations).make());
+    }
+
+    private boolean isHighQuality(final VariantContext vc) {
+        if (vc.getGenotypes().size() == 1) {
+            //genotyping engine returns 0.01 for hom-ref SNPs, probably because of prior
+            final Genotype genotype = vc.getGenotype(0);
+            if (!genotype.hasPL()) {
+                return false;
+            }
+            if (genotype.isHomRef()) {
+                return false;
+            }
+            if (genotype.getPL()[0] == 0) {
+                return false;
+            }
+            if (genotype.getAlleles().contains(Allele.NON_REF_ALLELE)) {
+                return false;
+            }
+            return vc.getPhredScaledQual() > 0.01 && (MathUtils.arrayMax(genotype.getPL()) > 0);
+        } else {
+            for (final Genotype genotype : vc.getGenotypes()) {
+                if (genotype.isHomRef()) {
+                    continue;
+                }
+                if (passesGnomadAdjCriteria(genotype)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+
+    //https://gnomad.broadinstitute.org/news/2020-10-gnomad-v3-1-new-content-methods-annotations-and-data-availability/
+    private boolean passesGnomadAdjCriteria(final Genotype g) {
```

Comment:
This is for known significant changes so we can compare only high quality genotypes as defined by these gnomAD adj criteria.